src/Agent.Worker/TestResults/ResultsCommandExtension.cs

```diff
@@ -104,15 +114,18 @@ private void ProcessPublishTestResultsCommand(IExecutionContext context, Diction
             var publisher = HostContext.GetService<ITestRunPublisher>();
             publisher.InitializePublisher(context, connection, teamProject, resultReader);
 
+            var ciService = HostContext.GetService<ICustomerIntelligenceServer>();
+            ciService.Initialize(connection);
+            
             var commandContext = HostContext.CreateService<IAsyncCommandContext>();
             commandContext.InitializeCommandContext(context, StringUtil.Loc("PublishTestResults"));
             if (_mergeResults)
             {
-                commandContext.Task = PublishAllTestResultsToSingleTestRunAsync(_testResultFiles, publisher, buildId, runContext, resultReader.Name, context.CancellationToken);
+                commandContext.Task = PublishAllTestResultsToSingleTestRunAsync(_testResultFiles, publisher, buildId, runContext, resultReader.Name, ciService, context.CancellationToken);
```

Comment:
why do we need to pass CI Service to this method?
If we are storing properties at class level, what's the usecase to pass this as method arg!?

Reply:
To remove ciService , i will need VssConnection for initializing it
and WorkerUtilities.GetVssConnection seems to create new connection whenever called.

so either i can pass ciService or connection for publishing
since this method doesn't take connection in the input..

Reply:
why this has to be passed to method? you are populating telemetry in the main method, why not just add _executionContext.AsyncCommands.Add(PublishTelemetryTask)?

Reply:
Ohh i though we could execute only single task. Will modify

Reply:
PublishTestResultsAsync(){
  if(_merged){ 
         await PublishAll 
  else  
         await PublishSingle
  await PublishTelemetry
}

CommandContext.task = PublishTestResultsAsync

Reply:
modified